test/semantics/call03.f90

```diff
@@ -0,0 +1,305 @@
+  subroutine test06 ! 15.5.2.4(4)
+    character :: ch1
+    ! ERROR: Length of effective character argument is less than required by dummy argument
+    call ch2(ch1)
+    ! ERROR: Length of effective character argument is less than required by dummy argument
+    call ch2(' ')
+  end subroutine
+
+  subroutine out01(x)
+    type(alloc) :: x
+  end subroutine
+  subroutine test07(x) ! 15.5.2.4(6)
+    type(alloc) :: x[*]
+    ! ERROR: coindexed effective argument with ALLOCATABLE ultimate component must be associated with a dummy argument with VALUE or INTENT(IN) attributes
+    call out01(x[1])
+  end subroutine
+
+  subroutine test08(x) ! 15.5.2.4(13)
+    real :: x[*]
+    ! ERROR: a coindexed scalar argument must be associated with a scalar dummy argument
+    call assumedsize(x[1])
+  end subroutine
+
+  subroutine charray(x)
+    character :: x(10)
+  end subroutine
+  subroutine test09(ashape, polyarray, c) ! 15.5.2.4(14), 15.5.2.11
+    real :: x, arr(10)
+    real, pointer :: p(:)
+    real :: ashape(:)
+    class(t) :: polyarray(*)
+    character(10) :: c(:)
+    ! ERROR: whole scalar argument cannot be associated with a dummy argument array
+    call assumedsize(x)
+    ! ERROR: element of pointer array cannot be associated with a dummy argument array
+    call assumedsize(p(1))
+    ! ERROR: element of assumed-shape array cannot be associated with a  dummy argument array
+    call assumedsize(ashape(1))
+    ! ERROR: element of polymorphic array cannot be associated with a dummy argument array
+    call poly(polyarray(1))
+    call charray(c(1:1))  ! not an error if character
+    call assumedsize(arr(1))  ! not an error if element in sequence
+    call assumedrank(x)  ! not an error
+    call assumedtypeandsize(x)  ! not an error
+  end subroutine
+
+  subroutine test10(a) ! 15.5.2.4(16)
+    real :: scalar, matrix
+    real :: a(*)
+    ! ERROR: rank of effective argument (0) differs from assumed-shape dummy argument (1)
+    call assumedshape(scalar)
+    ! ERROR: rank of effective argument (2) differs from assumed-shape dummy argument (1)
+    call assumedshape(matrix)
+    ! ERROR: assumed-size array cannot be associated with assumed-shape dummy argument
+  end subroutine
+
+  subroutine test11(in) ! C15.5.2.4(20)
+    real, intent(in) :: in
+    ! ERROR: effective argument associated with INTENT(OUT) dummy must be definable
+    call intentout(in)
+    ! ERROR: effective argument associated with INTENT(OUT) dummy must be definable
+    call intentout(3.14159)
+    ! ERROR: effective argument associated with INTENT(OUT) dummy must be definable
+    call intentout(in + 1.)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(in)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(3.14159)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(in + 1.)
```

Comment:
Maybe add a test for a parenthesized name:
```
call intentout(x)  ! OK
call intentout((x))  ! ERROR
```

Reply:
Yes.